### PR TITLE
Update classes to be PSR-4 compliant

### LINKED ISSUE #103: Make classes PSR-4 compliant
### Description

The classes in this module are not PSR-4 compliant, this can lead to some issues with autoloading.

Specifically...
- composer.json `autoload` entry points to the `code/` directory when it should point to `src/`.
- classes in `tests/SegmentFieldModifier` do not include `SegmentFieldModifier` in their namespace. 

### Additional context or points of discussion

_No response_

### Validations

- [X] You intend to implement the feature yourself
- [X] You have read the [contributing guide](https://docs.silverstripe.org/en/contributing/code/)
- [X] You strongly believe this feature should be in core, rather than being its own community module
- [X] You have checked for existing issues or pull requests related to this feature (and didn't find any)

## PRs
- https://github.com/silverstripe/silverstripe-segment-field/pull/104

### PR DESCRIPTION
**Pull request by Cambis**
## Description

- Update `autoload` configuration in `composer.json`
- Update namespacing for classes in `tests/SegmentFieldModifier`
- Add .gitattributes file

## Issues

- #103 

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -30,10 +30,20 @@
...
-    "prefer-stable": true
-}
\ No newline at end of file
+    "prefer-stable": true,
+    "config": {
+        "allow-plugins": {
+            "composer/installers": true,
+            "silverstripe/vendor-plugin": true
+        }
+    }
```
Comment (ChloeSartorelli) [inline composer.json]:
```suggestion
```
This isn't necessary for modules

File: composer.json [hunk 2]
```diff
@@ -30,10 +30,20 @@
     },
     "autoload": {
         "psr-4": {
-            "SilverStripe\\Forms\\": "code/",
-            "SilverStripe\\Forms\\Tests\\": "tests/"
+            "SilverStripe\\Forms\\": "src/"
+        }
+    },
+    "autoload-dev": {
```
Comment (ChloeSartorelli) [inline composer.json]: `autoload-dev` will cause problems when running tests locally if the module was installed into the vendor directory of an existing project (which is how the CMS Squad works most of the time).
It also prevents us from running these tests in conjunction with other modules e.g. as part of recipe-kitchen-sink.

Please revert this change.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
I've added the pull request template back into your PR description (it should have been there before - did you delete it?)
Please fill it in completely including ticking any checkboxes that apply.

**Comment by Cambis** [comment]
Addressed the changes @GuySartorelli, apologies this is my first time contributing to silverstripe. Still very excited :smiley:

**Comment by GuySartorelli** [comment]
No apology necessary, you've done great! Thanks for submitting this. Just letting ci run, if it goes green I'll be happy to merge this in.

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**